### DISCUSSION TITLE
Addons Discussion

### DISCUSSION THREAD

**Discussion by CrossVas** [VintageModsReforged/BlockHelperAddons#2] [Ideas]
This place is for discussing addons being developed for [BlockHelper](https://www.curseforge.com/minecraft/mc-mods/block-helper) mod.
Images, suggestions, any reasonable request, etc. All goes in here.

**Comment by CrossVas** [comment]
# GregTech Intergration

Started working on GregTech integration
![image](https://github.com/CrossVas/BlockHelperAddons/assets/25727222/f1320a6f-f48a-4a31-a8fd-fc4af33aa5a4)
![image](https://github.com/CrossVas/BlockHelperAddons/assets/25727222/baa3b073-f1fc-4f93-acb8-21108b746ba6)

**Reply by CrossVas** [reply]
More progress
![image](https://github.com/CrossVas/BlockHelperAddons/assets/25727222/0f444b80-e432-460d-ba8d-fd748ca22588)
![image](https://github.com/CrossVas/BlockHelperAddons/assets/25727222/7dc9b297-ab64-4c70-82d9-8782bba19e7d)

**Comment by Pyromanc3r** [comment]
I don't know how I managed to miss your post and the mod archive. Only now I saw it. Thank you so much for the MPS! I recently found it in the old modpack "FTB Unhinged", but there was an old version of it. I'm currently updating that modpack for myself. In that old version (0.7.0-534) there was a bug and it was impossible to craft "Ion Thruster". I had to fix it through MineTweaker.
![2024-05-02_00 30 48](https://github.com/CrossVas/BlockHelperAddons/assets/17619385/b6cb54b2-b295-456b-aa67-35427f6683a8)

Everything is fine with your version.
![2024-05-02_00 43 35](https://github.com/CrossVas/BlockHelperAddons/assets/17619385/07bdc184-63f6-4125-a3d7-e2ec15804344)

MineTweaker also helped me to add wood to the OreDict from Thaumcraft 3. That worked. Your Adv. Chainsaw cutting down all the wood (tree cap. mode is ON in config). It just seems to me that there is no need for another the Tree Capitator mod now. I think it makes more sense that way. We use primitive tools in the old-fashioned way, and with the appearance of chainsaws we can cut down the whole tree at once and it's cool. We get a kind of technological progression :D

Another thing is the question of balance. Because we have two versions of Advanced Chainsaw (GregTech and Gravisuite). The GregTech version is more difficult to create, but inferior to the Gravisuite version. I realize this is not a problem with this mod. Gravisuite is more of an add-on for IC2, not for GregTech. Of course, I can make the recipe I want via MineTweaker. I think it would be nice if your mod integrated with GT3 and used its materials. Or had a separate settings in the config file for complex Greg-mode. Mine Factory Reloaded and Modular Powersuits mods do that. Just a thought on the matter.

One more thing. I noticed that GregTech integrates with the ExtrabiomesXL mod. Biomes O' Plenty (I love BoP :)) has almost the same biomes and blocks. 
There is a Mesa biome in both mods.
![2024-05-02_02 09 54](https://github.com/CrossVas/BlockHelperAddons/assets/17619385/24de978c-7e77-4c3a-82eb-61dd10ea2f14)
![2024-05-02_01 04 05](https://github.com/CrossVas/BlockHelperAddons/assets/17619385/9e12e122-3ff8-4ca8-880d-e07083c6c843)

Red Rock from ExtrabiomesXL can be recycled into a useful resources
![2024-05-02_01 00 18](https://github.com/CrossVas/BlockHelperAddons/assets/17619385/278e8d9b-b511-4d38-8f47-20f9f3843927)

There's also Cracked Sand from the Wasteland biome. This too can be recycled in a centrifuge.
![2024-05-02_02 48 01](https://github.com/CrossVas/BlockHelperAddons/assets/17619385/88c2db7b-6af0-447d-9965-9cf9c3a013f6)

BoP has Wasteland as well. Except for the Cracked Sand. It seems to have been replaced by the Dried dirt block
![2024-05-02_01 12 26](https://github.com/CrossVas/BlockHelperAddons/assets/17619385/e285758a-e47a-4b2f-bb85-54fc6703b4e2)
![2024-05-02_01 02 43](https://github.com/CrossVas/BlockHelperAddons/assets/17619385/8f137201-e17a-4f99-88b0-3ef373027df1)

I don't know, maybe it makes sense for you to add this integration? Like through your "GTCompanion"? Since that mod is designed to bring back missing features. MineTweaker helped me and I made similar recipes for myself. And this way you could do away with MineTweaker altogether.
Just note, if you want to integrate these recipes, that Dried dirt is a furnace recipe (I disabled the crafting of this block. It too ugly to use anywhere anyway).


I just have a few off-topic questions (hope I'm not boring you? lol). Do you know how to get rid of those black spots?
![2024-05-02_01 31 49](https://github.com/CrossVas/BlockHelperAddons/assets/17619385/8082b476-8a6b-4ec4-9bc1-5d93d6f7fb19)

It's some kind of lighting bug, but I haven't found a solution. I googled and it seems like a lot of people have encountered this. I tried installing the fix from GregTech's website https://gregtech.overminddl1.com/downloads/gregtech_1.5.2/index.html "NoLightUpdateLagMC152.zip", but it doesn't seem to work. I'm using MultiMC (edit instance -> version -> Add to minecraft.jar). Fix for disabling infinite water is working.

And in general, I'm curious why you decided to make some mods for such an old version like 1.5.2? It's certainly a great thing to do. I was surprised that these older versions of the game are still supported. I've missed a lot of cool mods and I'm curious what they used to be like (the other big problem is finding mods for older versions of Minecraft). I'm not squeamish about playing an older version of MC. Many mods have lost some features in newer versions of the game and it's interesting to see what it used to be like. Now it's the turn to GT3 (I played a little bit GT5) and I'm very lucky to find such good auxiliary mods. Thank you! You've helped a lot already 😃

**Reply by CrossVas** [reply]
Unfortunately, the light fix mod doesn't work for me either, but I don't really mind having it. I do understand that it's an old version and there are bugs and moments that can't be fixed/changed.
![image](https://github.com/CrossVas/BlockHelperAddons/assets/25727222/529c8dc9-0f65-4f98-a893-b6c94183d911)

About balancing recipes, this would be nice to discuss on mod specific github page. This discussion is only for BlockHelperAddons.
Same for suggestions. Use bellow links for specific mod suggestion. Will look into it once I have time.
https://github.com/CrossVas/GraviSuiteReforged and https://github.com/CrossVas/GregTechCompanion-1.5.2

Now to answer your question regarding me making these mods for such an old version. 
I've been playing GT a lot, GT3 being my favorite, 1.5.2 is the version I play every now and then. Of course, with the new version having more and more mods like Jade/TOP/Waila etc. I wanted to make my time on 1.5.2 feel same or at least to be as close to that as possible. And given the fact that there is a waila-like mod, I decided to give it a try. Now I do enjoy my play time even more.
VintageHopperDiet was done for the same reason. It has some limitations tho, yet it's working as intended and that's enough (yet again, it's an old version, that's ok).
GraviSuite is more complex and require more knowledge on these old versions, but I'm slowly getting there. I did reforge GraviSuite 1.4.7 for 1.19.2 and I like how it turned out. Having `voldeloom` for 1.5.2 available, I decided to reforge it for 1.5.2 as well, adding new features like `Torch Placement`, `Vein Mining`, `Levitation`, etc, from 1.7.10 and above.

**Comment by Pyromanc3r** [comment]
I was thinking more about IC2 crops. Is it possible to add the growth stages of plants to the hint besides their characteristics?
Something like that:
![изображение](https://github.com/user-attachments/assets/25bdeb94-dd7c-407e-9a59-2fe195ec2ceb)

**Reply by CrossVas** [reply]
I've been thinking about it myself.
I'll see what I can display additionally.

I'm trying to follow this format: 
![image](https://github.com/user-attachments/assets/38bf3adb-2488-4ab1-a0c2-11c93b7806b9)

The `Stage/Size` and maybe the `Growth` percentage.

Will be added in the next update.

**Reply by CrossVas** [reply]
And this is what I've got: 
Left: when not fully grown
Right: when fully grown
![image](https://github.com/user-attachments/assets/b9d34550-0ca8-45fc-9fa2-0661cc8e82aa)

**Reply by Pyromanc3r** [reply]
I think it's pretty cool. 👍

**Reply by Pyromanc3r** [reply]
Where can I read about the internal mechanics of these IC2 crops?
I'm curious as to what these "Stress" and "Environment" section type values are. Are they some type of growth bonuses as I understand it?

**Reply by CrossVas** [reply]
Unfortunately, there is no much info about crops.

`Stress` in this case is the same `plant stress` from real life botany. Any unfavorable condition or substance that affects or blocks a plant's metabolism, growth, or development. Was added for display only in version `1.19.2`, but it was there hidden for a long time. I decided to display it too.
What it does and how to manage? 
The rule is simple. The less stress, the better. If the stress value goes above its max value (yes, it can go above `maxValue`) and stays there for a long time, the crop will die. You need to keep it lower than the max value. 
How to achieve low stress? 
Here is where `Environment` section comes in (it defines the stress value). The higher Env stats, the better. Same with fertilizer, and water from `Storage` section. `WeedEx` is optional. 
Environment is about the biome the crop is placed. More about it here: https://gtnh.miraheze.org/wiki/IC2_Crops_Guide#Nutrients,_Humidity_and_Air_Quality some of this info still applies to 1.4.7/1.5.2

| Biome| Humidity | Nutrients |
|:----------------|:------:|:------:|
| River       |2| 0 |
| Swampland |2| 2 |
| Forest    |1|1|
| ForestHills |1|1|
| Jungle|1|2|
| JungleHills|1|2|
| Desert|-1|0|
| DesertHills|-1 |0|
| MushroomIsland|0|2|
| MushroomIslandShore|0|2|

`Env. Quality` or `Air Quality` depends on the height the crop is placed. The higher, the better. Also it accounts for the clear sky. 
`Nutrients` are defined by the amount of subsoil it has underneath. Placing dirt is recommended.
`Humidity` depends on water and the farmland. If the farmland is `hydrated` it gives additional points.

**Reply by Pyromanc3r** [reply]
Oh, that's very useful information. Thanks.
I used to do some plant breeding and I noticed that sometimes some rare crops would just disappear, which was very annoying and I couldn't understand why it was happening. I guess it was this case with hidden conditions not fulfilled =\

**Reply by CrossVas** [reply]
Also, from my personal experience, crops with stats above 23 are more aggressive against each other, meaning higher chance to kill adjacent crops, spawn weed and ruining the whole field.
A good balance would be: `23 (Growth), 23 (Gain), 0 (Resistance)`

**Reply by CrossVas** [reply]
Also, for better management: https://github.com/VintageModsReforged/AdvancedCropnalyzer-1.5.2/releases

**Comment by Pyromanc3r** [comment]
I made a translation for a couple of mods (although everything is clear there). But I didn't manage to test it, because localisation is not loaded (not supported).
[loc.zip](https://github.com/user-attachments/files/18115785/loc.zip)

I also reworked the textures for `IronBarrels` (the current ones seemed boring to me). If you like it, you can include them in a future version. 
![2024-12-12_19 57 06](https://github.com/user-attachments/assets/550fff96-c830-4497-b3af-4dbeab939b13)

upper side:
![2024-12-12_19 36 16](https://github.com/user-attachments/assets/a00ffcea-619e-49ca-85ea-edc065d41407)

But these barrels lack sides. It would be nice to have that look:
![2024-12-12_20 57 18](https://github.com/user-attachments/assets/2ab240c7-8c77-429e-82c6-0503a1a9aef5)

**Reply by CrossVas** [reply]
Lang files are supported:
- create a new folder inside `config` folder called `<modid>/lang` and place your lang file there.
- go to mod config and find the `Supported Localizations` section (might be called differently, not sure) and list your file name there. 
If done correctly, you'll see this in the log:
```
2024-12-13 08:04:42 [INFO] [Vintage Suite] Trying to load ru_RU.lang file from /mods/vintagebarrels/lang folder...
2024-12-13 08:04:42 [INFO] [Vintage Suite] Couldn't load ru_RU.lang file from mods/vintagebarrels/lang folder for vintagebarrels... Did you put it in config/vintagebarrels/lang folder? Let me check...
2024-12-13 08:04:42 [INFO] [Vintage Suite] Trying to load ru_RU.lang file from /config/vintagebarrels/lang folder...
2024-12-13 08:04:42 [INFO] [Vintage Suite] Loaded lang file for ru_RU!
```

**Reply by CrossVas** [reply]
Textures look too dark to me. I don't like that.
I've made them as simple as possible in terms of textures, but if you could make your textures brighter, I might add them to the mod.
I do like the look actually:
![image](https://github.com/user-attachments/assets/ccf7b8b4-e6be-49ff-9fb3-0e7c77294d80)

**Reply by CrossVas** [reply]
Also, once again, can you report bugs to mod specific repo, so it's easier for me to keep track of it?

**Reply by CrossVas** [reply]
This looks too dark to me
![image](https://github.com/user-attachments/assets/b58ebb3e-6ecd-4c5f-b2d6-086297b869fb)

**Reply by CrossVas** [reply]
I'll submit the new texture format, please use those as reference if you want to re-work them.
Would be nice to have symmetric textures.
This doesn't look good
![image](https://github.com/user-attachments/assets/31c64917-4faf-40cb-ac6b-e18a935be2f3)

**Reply by CrossVas** [reply]
Also, what minimap are you using?
![image](https://github.com/user-attachments/assets/5cb735e9-ec6d-4493-ab3a-a9ef988e1cdb)

**Reply by Pyromanc3r** [reply]
>Also, what minimap are you using?

Sorry I didn't reply right away, been busy for a while.
This is ZansMinimap. As far as I understand in the future it became VoxelMap. I just changed the design of the map frame for my modpack and added there icons of different mobs from random mods. I use it together with MapWritter for full screen map.
[ZansMinimap1.5.2.zip](https://github.com/user-attachments/files/18156546/ZansMinimap1.5.2.zip)
[mapwriter-2.0.5.zip](https://github.com/user-attachments/files/18156547/mapwriter-2.0.5.zip)
